packages/runtime/container-runtime/src/test/garbageCollection.spec.ts

```diff
@@ -751,27 +757,36 @@ describe("Garbage Collection Tests", () => {
          * to GC.
          */
         it(`Scenario 7 - Reference added without notifying GC`, async () => {
-            // Initialize node A.
-            defaultGCData.gcNodes["/"] = [ nodeA ];
+            // Initialize nodes A & D.
+            defaultGCData.gcNodes["/"] = [ nodeA, nodeD ];
             defaultGCData.gcNodes[nodeA] = [];
+            defaultGCData.gcNodes[nodeD] = [];
 
             // 1. Run GC and generate summary 1. E = [].
             const timestamps1 = await getUnreferencedTimestamps();
             assert(timestamps1.get(nodeA) === undefined, "A should be referenced");
+            assert(timestamps1.get(nodeD) === undefined, "D should be referenced");
 
-            // 2. Create node B. E = [].
+            // 2. Create nodes B & C. E = [].
             defaultGCData.gcNodes[nodeB] = [];
+            defaultGCData.gcNodes[nodeC] = [];
 
-            // 3. Add reference from A to B without calling addedOutboundReference. E = [A -> B].
-            defaultGCData.gcNodes[nodeA] = [ nodeB ];
+            // 3. Add reference from A to B, A to C, and D to C without calling addedOutboundReference.
+            // E = [A -> B, A -> C, D -> C].
+            defaultGCData.gcNodes[nodeA] = [ nodeB, nodeC ];
+            defaultGCData.gcNodes[nodeD] = [ nodeC ];
 
-            // 4. Run GC and generate summary 2. E = [A -> B].
+            // 4. Run GC and generate summary 2. E = [A -> B, A -> C, D -> C].
             await getUnreferencedTimestamps();
 
-            // Validate that we got the "gcUnknownOutboundRoute" error.
-            mockLogger.matchEvents([
-                { eventName: unknownRouteEvent, route: nodeB },
-            ]);
+            // Validate that we got the "gcUnknownOutboundReferences" error.
+            const unknownReferencesEvent = "GarbageCollector:gcUnknownOutboundReferences";
+            const eventsFound = mockLogger.matchEvents([{
+                    eventName: unknownReferencesEvent,
+                    gcNode: "/",
+                    gcRoutes: JSON.stringify(["/A", "/D"]),
```

Comment:
This should have A -> B, A -> C and D -> C since those were added between the two GC runs without notifying GC.